src/app/view/components/dot-navigation/dot-navigation.service.spec.ts

```diff
@@ -0,0 +1,208 @@
+        spyOn(dotMenuService, 'isPortletInMenu').and.returnValue(Observable.of(true));
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(dotNavigationService, 'goToFirstPortlet');
+
+        const mockUser = {
+            emailAddress: 'admin@dotcms.com',
+            firstName: 'Admin',
+            lastName: 'Admin',
+            loggedInDate: 123456789,
+            userId: '123',
+            editModeUrl: '/test/123'
+        };
+        const mockAuth: Auth = {
+            loginAsUser: null,
+            user: mockUser
+        };
+        loginService.triggerNewAuth(mockAuth);
+
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('edit-page/content?url=/test/123', true);
+        expect(dotNavigationService.goToFirstPortlet).not.toHaveBeenCalled();
+    });
+
+    it('should redirect to previous saved URL and clear the saved url', fakeAsync(() => {
+        spyOn(dotMenuService, 'reloadMenu').and.returnValue(Observable.of(mockMenu));
+        spyOn(dotMenuService, 'isPortletInMenu').and.returnValue(Observable.of(true));
+        spyOn(dotNavigationService, 'goToFirstPortlet');
+        spyOn(dotRouterService, 'gotoPortlet').and.callFake(() => {
+            return new Promise(resolve => {
+                resolve(true);
+            });
+        });
+        dotRouterService.previousSavedURL = 'hello/world';
+
+        loginService.triggerNewAuth(baseMockAuth);
+
+        tick();
+        expect(dotNavigationService.goToFirstPortlet).not.toHaveBeenCalled();
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('hello/world', true);
+        expect(dotRouterService.previousSavedURL === null).toBe(true);
+    }));
+
+    // TODO: needs to fix this, looks like the dotcmsEventsService instance is different here not sure why.
+    xit('should subscribe to UPDATE_PORTLET_LAYOUTS websocket event', () => {
+        spyOn(dotcmsEventsService, 'subscribeTo');
+        expect(dotcmsEventsService.subscribeTo).toHaveBeenCalledWith('UPDATE_PORTLET_LAYOUTS');
+    });
+
+    it('should go to first portlet', () => {
+        spyOn(dotMenuService, 'loadMenu').and.returnValue(Observable.of(mockMenu));
+
+        spyOn(dotRouterService, 'gotoPortlet');
+        dotNavigationService.goToFirstPortlet(false);
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('hello/url', false);
+    });
+
+    it('should return correct value in isActive', () => {
+        expect(dotNavigationService.isActive('hello')).toBe(true);
+    });
+
+    it('should reload current portlet', () => {
+        spyOn(dotRouterService, 'reloadCurrentPortlet');
+        dotNavigationService.reloadCurrentPortlet('hello');
+
+        expect(dotRouterService.reloadCurrentPortlet).toHaveBeenCalledWith('hello');
+    });
+
+    it('should NOT reload current portlet', () => {
+        spyOn(dotRouterService, 'reloadCurrentPortlet');
+        dotNavigationService.reloadCurrentPortlet('123');
```

Comment:
why do the reloadCurretPortlet method have a 'id' parameter?

Reply:
Because we compare if the url have the same "id" the it reloads.